distributed/scheduler.py

```diff
@@ -1590,6 +1596,36 @@ def get_nbytes(self, stream=None, keys=None, summary=True):
 
             return result
 
+    def publish_dataset(self, stream=None, keys=None, data=None, name=None,
+                        client=None):
+        for k in keys:
+            self.who_wants[k].add('published-%s' % name)
+        self.published_data[name] = {'data': data, 'keys': keys,
+                                     'clients': {client}}
+
+    def unpublish_dataset(self, stream=None, name=None):
+        out = self.published_data.pop(name, {'keys': []})
+        for k in out['keys']:
+            self.who_wants[k].remove('published-%s' % name)
```

Comment:
And actually I strongly recommend using the `client_releases_keys` function for all of this.  We want to centralize logic like this to avoid hard-to-find errors.